src/ddsrt/src/process/posix/process.c

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright(c) 2006 to 2018 ADLINK Technology Limited and others
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Eclipse Distribution License
+ * v. 1.0 which is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR BSD-3-Clause
+ */
+#include <stdio.h>
+#include <errno.h>
+#include <string.h>
+#include <unistd.h>
+#include <assert.h>
+#include <stdlib.h>
+#include <sys/types.h>
+#include "dds/ddsrt/process.h"
+
+
+ddsrt_pid_t
+ddsrt_getpid(void)
+{
+  /* Mapped to taskIdSelf() in VxWorks kernel mode. */
+  return getpid();
+}
+
+
+#ifdef DDSRT_USE_PROCESSCREATION
```

Comment:
Can you explain this a bit more? I don't really understand this remark.

Reply:
Removed.

Reply:
@k0ekk0ek I think I understand what you were trying to say, but I have to agree with @martinbremmer that it is a pretty confusing statement.

This:

> But, since all supported platforms support it, there's no need to check here.

however, is the wrong way to look at it: if you want a truly minimal build, you would want to leave it out, and secondly, there is no need for any process management other than for testing and so it is not needed for a port of Cyclone to a minimalistic platform. A compile-time option makes it clear that it isn't really required for Cyclone proper.

Reply:
That is not the way it is supposed to work. Whether or not it's supported is determined by the platform and that check is implicit. For instance, I'm currently doing a port to FreeRTOS. That platform contains no such thing as seperate processes. If a feature is not supported on all platforms, there should be a test in `src/ddsrt/cmake/<feature>.c`. That is checked by the build system and based on that information a feature is added or not. Of course, we could extend it to leave certain features out by specifying an option. In any case, don't call the option *DDSRT_USE_<feature>*. As it'll be confusing. Simply call it *WITH_PROCESS* or something. Once I do a pull request with my changes everything will become much clearer, because there'll be examples.

Reply:
Introduced WITH_PROCESS to be able to leave process management out of the code if you want a minimal build.